### DISCUSSION TITLE
Check my work? Blank screen with ST779 + ESP32

### DISCUSSION THREAD

**Discussion by MawiRecon** [Bodmer/TFT_eSPI#3213] [Q&A - General]
I'll preface this by saying that I'm new to PlatformIO/VS. I've been an Arduino IDE user for a while and completely comfortable in that IDE. 

_1. A description of the problem and the conditions that cause it to occur_
I'm trying to run a simple "Hello Word" print test on my TFT display in effort to get familiar with PlatformIO. 
The program compiles and uploads successfully, however the TFT remains blank. 
The issue is no doubt user error for updating User_Setup.h, however I can't figure out what the error is.
I've successfully tested this in Arduino IDE just to make sure the components are functioning properly.

I've added the TFT_eSPI lib to my project and started updating the User_Setup.h to align with my TFT and controller. Below are the relevant definitions, with everything else commented out. The pinouts are correct as this works using ArduinoIDE. 

```
#define USER_SETUP_INFO "User_Setup"
// ###### EDIT THE PIN NUMBERS IN THE LINES FOLLOWING TO SUIT YOUR ESP32 SETUP   ######
#define ST7789_DRIVER  
#define TFT_WIDTH  240
#define TFT_HEIGHT 320
#define TFT_CS   14 
#define TFT_DC    32 
#define TFT_RST   15
[... included the first 8 font definitions...] 
#define SPI_FREQUENCY  40000000
#define SPI_READ_FREQUENCY  20000000
```  

main.cpp is a simple "Hello World" print test:
```
#include <Arduino.h>
#include <TFT_eSPI.h>
#include <SPI.h>
TFT_eSPI tft = TFT_eSPI(); 

void setup() {
  Serial.begin(115200);
  tft.begin(); // Initialize the TFT display
  tft.setRotation(1); // Adjust the rotation if needed
  tft.fillScreen(TFT_BLACK); // Clear the screen with black color
  tft.setTextColor(TFT_WHITE); // Set text color to white
  tft.setTextSize(2); // Set text size
  tft.setCursor(10, 10); // Set the cursor position
  tft.print("Hello, World!"); 
}
void loop() {
}
```
I've tried some of the example programs as well, still with no luck. Just a blank screen. 

_2. IDE (e.g. Arduino or PlatformIO)_
PlatformIO

_3. TFT_eSPI library version_
2.5.43

_4. Board package version_ 
Using the adafruit_feather_esp32_v2 board 

_5. Procesor_
ESP32-PICO-MINI-02 
https://learn.adafruit.com/adafruit-esp32-feather-v2/pinouts

_6. TFT driver_ 
ST7789
https://www.adafruit.com/product/4311#technical-details

_7. Interface type (SPI or parallel)_
SPI

Plus further information as appropriate to the problem:
1. TFT to processor connections used
Using an 18pin EYESPI breakout to connect the TFT to the ESP32
https://www.adafruit.com/product/5613

**Comment by HapCoderWei** [comment]
Hi MawiRecon, how do you add TFT_eSPI lib to your PIO project, can you take a screenshot of the workspace of your project PIO? The following screenshot is my PIO's workspace. You can see the location of the User_Setup.h file.
And a small suggestion, when using the TFT_ESPI library for the first time, setting the background of the TFT to bright colors is easier to find the problem.
```
void setup() {
  tft.init();
  tft.setRotation(1);
  tft.fillScreen(TFT_RED);
}
```

![PIO_Project](https://github.com/Bodmer/TFT_eSPI/assets/6915031/6e2063ad-df5b-4e50-b27e-b297d2fd6ec5)
